tensorflow/python/distribute/input_lib_test.py

```diff
@@ -1250,5 +1250,99 @@ def sum_while_loop(ds):
     self.assertAllEqual(nest.flatten(sums), [expected_for_sum] * 3)
 
 
+class DistributedIteratorPerDeviceTest(DistributedIteratorTestBase, parameterized.TestCase):
+  """Tests for PER_WORKER and PER_REPLICA's InputOptions variants."""
+
+  def setUp(self):
+    context._reset_context()
+    strategy_combinations.set_virtual_cpus_to_at_least(3)
+    super(DistributedIteratorPerDeviceTest, self).setUp()
+
+  @combinations.generate(
+      combinations.combine(
+          input_options=[
+              distribute_lib.InputOptions(
+                  experimental_place_dataset_on_device=False,
+                  experimental_prefetch_to_device=True,
+                  experimental_replication_mode=distribute_lib.InputReplicationMode.PER_WORKER),
+              distribute_lib.InputOptions(
+                  experimental_place_dataset_on_device=False,
+                  experimental_prefetch_to_device=True,
+                  experimental_replication_mode=distribute_lib.InputReplicationMode.PER_REPLICA),
+          ],
+          mode=["eager"],
+          distribution=[
+              strategy_combinations.mirrored_strategy_with_two_gpus,
+              strategy_combinations.mirrored_strategy_with_cpu_1_and_2,
+              strategy_combinations.mirrored_strategy_with_gpu_and_cpu,
+          ]))
+  def testInputSignatureForPerWorkerValuesWithPrefetch(self, distribution, input_options):
```

Comment:
the tests are not testing InputSignature etc, please rename appropriately. Also this test is testing both PER_WORKER and PER_REPLICA

Reply:
addressed!